### ISSUE TITLE
Can we integrate ISSO in SIngle Page web app like Angular 4

### ISSUE THREAD

**Issue by geovlazar** [posativ/isso#345]
Hi 

Nowadays lots of single-page web applications are created. We have migrated our application with Angular4 and its normal page structure is like sitename.com/#/page-name. But ISSO comments takes different pages after hash. Is there any way to treat this different content pages as separate URL under Angular 
Any help is appreciated

**Comment by claudio4** [comment]
You need I.E. support?

**Comment by sukima** [comment]
Ping? Any up date to this. The ability to place multiple threads per single page is a must. I am evaluation commenting systems that would be compatible with TiddlyWiki.com which shows each post as a stream of cards so multiple blog posts would be in one DOM. Can ISSO handle that?

**Comment by blatinier** [comment]
Sorry for the poor reactivity. I don't know if it works but there might be a possibility that it does.
Threads are based on URLs but you can specify a `data-isso-id` to bypass this. However the thread itself has an attribute `id="isso-thread"` so I suspect that multiple threads at the same time on a page might fail but one thread on a SinglePageApp might work.
No sure thing since I didn't tested any of those. Any feedback on these two usecases is very welcome!

**Comment by geovlazar** [comment]
Thanks, @blatinier . Need to dig further into this and will update

**Comment by sukima** [comment]
IIRC looking at the source. the API supports separate threads through the `isso-id` tag. It is the client side script that make assumptions on its use in a web page. Most static sites separate articles as individual pages allowing a script like ISSO and Disqus to presume that the whole page (URL) is a unique identifier. Even many SPA will manipulate the URL to reflect a unique identifier.

In some cases (like in my case) my page has multiple articles on one page and through the SPA will show hide as needed. Therefore the solution is to use ISSO (not Disqus because their API doesn't support this while ISSO does) a custom JavaScript library that will take into account multiple comment threads on a single page.

The down side at the moment is that to do separate threads it requires a single AJAX request per thread. Add on counts and that is two AJAX requests per thread. If I have say 25 articles on display that make 50 AJAX requests total. Or if I hide the comment threads until the user clicks then it would be 25 AJAX requests for the counts and then one request per user click.

It would be awesome if the API supported collating. Perhaps a site wide count API that returns a hash of topic ids to counts like `GET /api/all?url=www.example.com`:

```json
[
  {
    "titile": "Lorem",
    "count": 4,
    "thread_url": "http://www.example.com/api/thread?isso-id=12345"
  },
  ...
]
```

And then the SPA could add buttons and labels were needed.

**Comment by rocka** [comment]
Maybe you would be intersted in #452 ...

**Comment by StructByLightning** [comment]
In case anyone else finds this, I got it working on my SPA with the following steps:

Whenever the page changes:
1. Create a new script tag with all the isso data (reply-to-self, vote-levels, etc). Set its src to embed.min.js.
2. Remove the old #isso-thread.
3. Create a new #isso-thread with attribute "data-isso-id" set to a unique ID for the new page.
4. Inject the new script tag. Isso will update automatically and insert the new page's comments into #isso-thread.

**Comment by blatinier** [comment]
I merged #452 which provides necessary code & doc to integrate into SinglePageApps

**Comment by Shubhamsharma585** [comment]
Yes, I achieved this, you can create your own script which will create a div with classname 'dynamic-div'  and insert it dynamically.  It tested from my side and worked perfectly just like many forms in a single page
